pallets/did/src/tests.rs

```diff
@@ -1670,6 +1679,81 @@ fn check_too_large_tx_counter_call_error() {
 	});
 }
 
+#[test]
+fn check_tx_block_number_too_low_error() {
+	let auth_key = get_sr25519_authentication_key(true);
+	let did = get_did_identifier_from_sr25519_key(auth_key.public());
+	let caller = ACCOUNT_00;
+
+	let mock_did = generate_base_did_details::<Test>(did::DidVerificationKey::from(auth_key.public()));
+
+	let mut ext = ExtBuilder::default()
+		.with_dids(vec![(did.clone(), mock_did)])
+		.build(None);
+
+	let submitter = kilt_primitives::AccountId::default();
+
+	let call_operation = generate_test_did_call(did::DidVerificationKeyRelationship::Authentication, did, submitter);
+	let signature = auth_key.sign(call_operation.encode().as_ref());
+
+	ext.execute_with(|| {
+		// System block number 1 past the max block the operation was allowed for.
+		System::set_block_number(call_operation.operation.block_number + MaxBlocksTxValidity::get() + 1);
+		assert_noop!(
+			Did::submit_did_call(
+				Origin::signed(caller.clone()),
+				Box::new(call_operation.operation.clone()),
+				did::DidSignature::from(signature.clone())
+			),
+			did::Error::<Test>::InvalidOperationValidity
+		);
+	});
+
+	// But it would work if the system would be one block earlier.
+	ext.execute_with(|| {
+		System::set_block_number(call_operation.operation.block_number + MaxBlocksTxValidity::get());
+		assert_ok!(Did::submit_did_call(
+			Origin::signed(caller),
+			Box::new(call_operation.operation),
+			did::DidSignature::from(signature)
+		),);
+	});
+}
+
+#[test]
+fn check_tx_block_number_too_high_error() {
+	let auth_key = get_sr25519_authentication_key(true);
+	let did = get_did_identifier_from_sr25519_key(auth_key.public());
+	let caller = ACCOUNT_00;
+
+	let mock_did = generate_base_did_details::<Test>(did::DidVerificationKey::from(auth_key.public()));
+
+	let mut ext = ExtBuilder::default()
+		.with_dids(vec![(did.clone(), mock_did)])
+		.build(None);
+
+	let submitter = kilt_primitives::AccountId::default();
+
+	let mut call_operation =
+		generate_test_did_call(did::DidVerificationKeyRelationship::Authentication, did, submitter);
+	call_operation.operation.block_number = MaxBlocksTxValidity::get() + 100;
+	let signature = auth_key.sign(call_operation.encode().as_ref());
+
+	ext.execute_with(|| {
+		// System block number is still too low, meaning that the block number used in
+		// the operation was too high.
+		System::set_block_number(call_operation.operation.block_number - MaxBlocksTxValidity::get() - 1);
+		assert_noop!(
+			Did::submit_did_call(
+				Origin::signed(caller.clone()),
+				Box::new(call_operation.operation.clone()),
+				did::DidSignature::from(signature.clone())
+			),
+			did::Error::<Test>::InvalidOperationValidity
+		);
+	});
```

Comment:
W: Wrap all of this into the execution of the Externality.

Reply:
https://github.com/KILTprotocol/mashnet-node/pull/257/commits/639389dc8c1f3317b4a523fa6225412a21d53b01